src/toil/server/wes/toil_backend.py

```diff
@@ -0,0 +1,335 @@
+            },
+            "supported_wes_versions": ["1.0.0"],
+            "supported_filesystem_protocols": ["file", "http", "https"],
+            "workflow_engine_versions": {"toil": baseVersion},
+            "default_workflow_engine_parameters": [
+                {
+                    "name": key,
+                    "default_value": value
+                }
+                for key, value in engine_parameters
+            ],
+            "system_state_counts": state_counts,
+            "tags": {},
+        }
+
+    @handle_errors
+    def list_runs(self, page_size: Optional[int] = None, page_token: Optional[str] = None) -> Dict[str, Any]:
+        """ List the workflow runs."""
+        # TODO: implement pagination
+        return {
+            "workflows": [
+                {
+                    "run_id": run_id,
+                    "state": state
+                } for run_id, state in self.get_runs()
+            ],
+            "next_page_token": ""
+        }
+
+    @handle_errors
+    def run_workflow(self) -> Dict[str, str]:
+        """ Run a workflow."""
+        run_id = uuid.uuid4().hex
+        job = self._get_run(run_id, should_exists=False)
+
+        # set up necessary directories for the run
+        job.set_up_run()
+
+        # stage the uploaded files to the execution directory, so that we can run the workflow file directly
+        temp_dir = job.exec_dir
+        try:
+            _, request = self.collect_attachments(run_id, temp_dir=temp_dir)
+        except ValueError:
+            job.clean_up()
+            raise
+
+        wf_type = request["workflow_type"].lower().strip()
+        version = request["workflow_type_version"]
+
+        # validate workflow request
+        supported_versions = self.supported_versions.get(wf_type, None)
+        if not supported_versions:
+            job.clean_up()
+            raise RuntimeError(f"workflow_type '{wf_type}' is not supported.")
+        if version not in supported_versions:
+            job.clean_up()
+            raise RuntimeError("workflow_type '{}' requires 'workflow_type_version' to be one of '{}'.  Got '{}'"
+                               "instead.".format(wf_type, str(supported_versions), version))
+
+        job.set_state("QUEUED")
+
+        command = job.get_run_command(request=request, options=self.options)
+
+        # TODO: Flask recommends using a task queue (like Celery or rq) to run tasks like these, since this way
+        #  will likely leave behind zombie child processes until we read the return code of the process.
+        # https://flask.palletsprojects.com/en/2.0.x/patterns/celery/
+        # https://python-rq.org/
+
+        with open(os.path.join(job.work_dir, "runner.log"), "w") as log:
+            process = subprocess.Popen(command, stdout=log, stderr=log, cwd=job.work_dir)
+
+        with open(os.path.join(job.work_dir, "runner.pid"), "w") as f:
+            f.write(str(process.pid))
```

Comment:
I believe Toil itself also generates a pid file.  We should probably reconcile these.